app/views/projects/_form_0.html.erb

```diff
@@ -209,8 +209,19 @@
                          details:
                            t('projects.form_basics.license.details_html')
                        }) %>
-            <%= f.text_field :license,
+            <%=
+                # "Pattern" here is not a rigorous check of SPDX syntax.
+                # It only checks to ensure that only legal characters are
+                # allowed. Still, this can prevent some problems we've seen
+                # in some submissions, while permitting legal expressions like
+                # "(GPL-2.0+ WITH Bison-exception-2.2)".  See:
+                # https://spdx.org/spdx-specification-21-web-version
+                # Errors here do not cause *security* issues, so we're not
+                # too strict.
+                f.text_field :license,
                              class: 'form-control', hide_label: true,
+                             pattern: '(|[()A-Za-z0-9\-.+ ]*)',
```

Comment:
This only fixes the input if done through a web form. I would recommend also implementing a model validation here: https://github.com/coreinfrastructure/best-practices-badge/blob/master/app/models/project.rb#L165